pkg/controller/networkpolicy/controller_query_test.go

```diff
@@ -104,20 +137,40 @@ var policies = []networkingv1.NetworkPolicy{
 var namespaces = []v1.Namespace{
 	{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:                       "testNamespace",
-			UID:                        "testNamespaceUID",
+			Name: "testNamespace",
+			UID:  "testNamespaceUID",
 		},
 	},
 }
 
-func makeControllerAndEndpointQueryReplier(objects ...runtime.Object) (*networkPolicyController, *EndpointQueryReplier){
+func makeControllerAndEndpointQueryReplier(objects ...runtime.Object) (*networkPolicyController, *EndpointQueryReplier) {
 	// create controller
-	_, controller := newController(objects...)
+	_, c := newController(objects...)
+	c.heartbeatCh = make(chan heartbeat, 1000)
+	stopCh := make(chan struct{})
 	// create querier with stores inside controller
-	address, appliedToGroup, policy, podInformer := controller.addressGroupStore, controller.appliedToGroupStore, controller.internalNetworkPolicyStore, controller.podInformer
-	querier := NewEndpointQueryReplier(address, appliedToGroup, policy, podInformer)
-
-	return controller, querier
+	querier := NewEndpointQueryReplier(c.NetworkPolicyController)
+	// start informers and run controller
+	c.informerFactory.Start(stopCh)
+	go c.Run(stopCh)
+	// wait until computation is done
+	idleTimeout := 3 * time.Second
```

Comment:
we can make that delay a parameter to the function: I think 3 seconds is an overkill for the regular unit tests for which we can try using 500ms (but is probably appropriate for the benchmark test).